wp-content/plugins/wporg-learn/inc/post-meta.php

```diff
@@ -521,6 +521,16 @@ function save_workshop_metabox_fields( $post_id ) {
 			add_post_meta( $post_id, 'other_contributor_wporg_username', $username );
 		}
 	}
+
+	// This language meta field is rendered in the editor sidebar using a PluginDocumentSettingPanel block,
+	// which won't save the field on publish if it has the default value.
+	// Our custom workshops query for locale prioritized tutorials (see functions.php `wporg_archive_query_prioritize_locale`)
+	// depends on it being set, so we force it to be updated after saving:
+	$language         = get_post_meta( $post_id, 'language', true );
+	$language_default = 'en_US';
+	if ( ! isset( $language ) || $language_default === $language ) {
```

Comment:
Forgive me if I'm reading this wrong, but is the $language_default === $language check required? 

if $language is already en_US, do we still need to update it?

Reply:
Yeah good spotting. When debugging I found that the meta field is read as the default value even if the value is not saved in the db. So in this case there is no post meta entry for `language` but $language = 'en_US', and so we still need to force save to actually add a post meta entry. It's definitely not ideal.